src/computation/logic/distribute-levelling-seats.ts

```diff
@@ -21,75 +100,124 @@ export function distributeLevelingSeats(
             votes: partyResults[partyCode].votes,
             percentage: -1,
         };
-    });
-
-    // Filter out parties with less votes than the threshold or who did not gain any seats from the national distribution
-    levelingParties = levelingParties.filter(
-        (party) => partyResults[party.partyCode].percentVotes >= payload.electionThreshold
-    );
-
-    let seatsToDistribute = payload.levelingSeats;
-    levelingParties.forEach((party) => {
-        seatsToDistribute += partyResults[party.partyCode].districtSeats;
-    });
+        levelingParties.push(party);
+    }
 
     // Compute the distribution of the total number of seats on the whole country
     const nationalDistribution = distributeSeats(
         payload.algorithm,
         payload.firstDivisor,
         Number.MIN_SAFE_INTEGER,
         seatsToDistribute,
+        totalVotes,
         levelingParties
     );
 
-    // Filter out parties who did not gain any seats from the national distribution
-    levelingParties = levelingParties.filter(
-        (party) => nationalDistribution.seatsWon[party.partyCode] >= partyResults[party.partyCode].districtSeats
+    // Filter out parties that did not gain any seats in the new distribution
+    const filteredLevelingPartyCodes = levelingPartyCodes.filter(
+        (p) => nationalDistribution.seatsWon[p] > partyResults[p].districtSeats
     );
 
-    let levelingPartyCodes = levelingParties.map((party) => party.partyCode);
+    if (filteredLevelingPartyCodes.length !== levelingPartyCodes.length) {
```

Comment:
The conditional here isn't very intuitive to me -- what does it actually represent?

Reply:
If we found a party that should be filtered out, the national distribution should be repeated. Here is an example from Stortinget which uses this behaviour: [https://www.stortinget.no/no/Saker-og-publikasjoner/Publikasjoner/Innstillinger/Stortinget/2005-2006/inns-200506-001/7/](https://www.stortinget.no/no/Saker-og-publikasjoner/Publikasjoner/Innstillinger/Stortinget/2005-2006/inns-200506-001/7/)

Reply:
const existsUnfilteredParty = ...conditional... perhaps?

Reply:
Fixed.

Reply:
Did you push the change you made? I can't see it.

Reply:
Pushed two changes now, see if it updated.